lib/rules/no-extra-boolean-cast.js

```diff
@@ -110,6 +159,10 @@ module.exports = {
                                 prefix = " ";
                             }
 
+                            if (astUtils.getPrecedence(node.argument) < astUtils.getPrecedence(parent.parent)) {
+                                return fixer.replaceText(parent, `(${prefix + sourceCode.getText(node.argument)})`);
```

Comment:
I think that `prefix` isn't necessary in this branch because fixer inserts parens anyway? An example:

```js
/*eslint no-extra-boolean-cast: ["error", {"enforceForLogicalOperands": true}]*/

function *foo() { 
    yield!!(a = b) || d ? b : c 
}
```

in this version would be fixed to:

```js
/*eslint no-extra-boolean-cast: ["error", {"enforceForLogicalOperands": true}]*/

function *foo() { 
    yield( a = b) || d ? b : c 
}
```

so there is an unnecessary space prefix before `a`.